ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiSnapshotWorldState.java

```diff
@@ -84,5 +92,24 @@ public void close() throws Exception {
     codeSnap.close();
     storageSnap.close();
     trieBranchSnap.close();
+    parentWorldStateStorage.unSubscribe(parentWorldStateSubscriberId);
+  }
+
+  @Override
+  public void onClear() {
+    try {
+      close();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
how we  manage this case  in the other level of the code ?

Reply:
all of the retrievals of SnapshotWorldStates are in a try-with-resources block, so there is implicitly a catch()

Reply:
this would result in an exception from the caller trying to clear/truncate.  Currently that only happens at the beginning of a fresh snap sync.  I will force it to happen to ensure that snap sync would restart. Really clear() is quite risky with snapshots.  It might be worth overriding it with an iterator-delete based approach if we are using snapshots.